scripts/template_matcher.py

```diff
@@ -58,16 +67,90 @@ def _compute_prediction(self, k, img, kp, des):
         k: template image for comparison, img: scene image
         kp: keypoints from scene image,   des: descriptors from scene image
         """
-
-        # TODO: find corresponding points in the input image and the templae image
-        #       put keypoints from template image in template_pts
-        #       put corresponding keypoints from input image in img_pts
-
-
-        #TODO: change img to img_T once you do the homography transform
-        visual_diff = compare_images(img, self.signs[k])
+        ###taken from example code:
+
+        #find matches
+        matches = self.matcher.knnMatch(des,self.descs[k],k=2)
+
+        # make sure the distance to the closest match is sufficiently better than the second closest
+        good_matches = []
+        for m,n in matches:      
+            if (m.distance < 0.7*n.distance and
+                kp[m.queryIdx].response > 0.0 and
+                self.kps[k][m.trainIdx].response > 0.0):
+                good_matches.append((m.queryIdx, m.trainIdx))
+        
+        if len(good_matches) < 4:
+            visual_diff = None
+        else:
+            #create pt lists
+            img_pts = np.zeros((len(good_matches),2))
+            tem_pts = np.zeros((len(good_matches),2))
+            for idx in range(len(good_matches)):
+                match = good_matches[idx]
+                img_pts[idx,:] = kp[match[0]].pt
+                tem_pts[idx,:] = self.kps[k][match[1]].pt
+        
+            # Transform input image so that it matches the template image as well as possible
+            M, mask = cv2.findHomography(img_pts, tem_pts, cv2.RANSAC, 8.0)
```

Comment:
Did you get any luck on using a ransac_threshold of 8.0 vs the default 5.0?   Just curious if you spent some time playing with this.